crates/lair_keystore/src/store_sqlite.rs

```diff
@@ -584,16 +625,29 @@ fn encrypt_unencrypted_database(
         conn.execute("BEGIN EXCLUSIVE", ())
             .map_err(one_err::OneErr::new)?;
 
-        let lock = key_pragma.read_lock();
-        conn.execute(
-            "ATTACH DATABASE :db_name AS encrypted KEY :key",
-            rusqlite::named_params! {
-                ":db_name": encrypted_path.to_str(),
-                ":key": &lock[14..81],
-            },
+        {
+            let lock = key_pragma.read_lock();
+            conn.execute(
+                "ATTACH DATABASE :db_name AS encrypted KEY :key",
+                rusqlite::named_params! {
+                    ":db_name": encrypted_path.to_str(),
+                    ":key": &lock[14..81],
+                },
+            )
+            .map_err(one_err::OneErr::new)?;
+        }
+
+        conn.execute_batch(
```

Comment:
There's a test for migration that will open the newly encrypted database with the same compatibility settings right?

Just thinking ahead to when we move versions and only remember to update one :)

Reply:
(and yes, there's a `migrate_unencrypted` test that triggers this code to be run.)